core/vulkan/perfetto_producer/perfetto_data_source.inc

```diff
@@ -91,6 +92,16 @@ void PerfettoProducer<T>::OnStop(const typename perfetto::DataSourceBase::StopAr
     Get().OnStop(a);
 }
 
+template<typename T>
+void PerfettoProducerData<T>::OnSetup(const typename perfetto::DataSourceBase::SetupArgs& a) {
+    emitter_lock_.Lock();
+    started_ = true;
+    for (auto e : emitters_) {
```

Comment:
Guess it doesn't matter so much because of the map's type, but auto&& is safer here to avoid copying data

Reply:
We probably dont want auto&&, since we really wouldn't want move semantics, auto& however is probably ok.